drivers/timer/cortex_m_systick.c

```diff
@@ -402,9 +402,20 @@ void sys_clock_idle_exit(void)
 		/* Calculate difference in measurements to get how much time
 		 * the SysTick missed in idle state.
 		 */
-		measurement_diff_us = idle_timer_us - systick_us;
-		missed_cycles =
-			(sys_clock_hw_cycles_per_sec() * measurement_diff_us) / USEC_PER_SEC;
+		if (idle_timer_us < systick_us) {
+			/* This case is possible, when the time in low power mode is
+			 * very short or 0. SysTick usually has higher measurement
+			 * resolution of than the IDLE timer, thus the measurement of
+			 * passed time since the sys_clock_set_timeout call can be higher.
```

Comment:
`systick_diff` is calculated from cycle counters `cycle_post_idle` and `cycle_pre_idle`, not SysTick register counter itself. The cycle counters are calculated with the `elapsed` function, which already handles the wrap.

Anyway, the wrap shouldn't happen anyway, because this code is used in case SysTick is stopped in idle. There is a short window when SysTick still works before and after idle, but just before `sys_clock_set_timeout` with idle=true, the `sys_clock_set_timeout` is called with idle=false (with a few more ticks). So the SysTick is reprogrammed with `load` > `min-residency`.

To sum up, I don't see a risk of the wrap and even if it happens, it is handled well.